src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala

```diff
@@ -466,8 +466,18 @@ class Checker(globalTable: SymbolTable, verbose: Boolean = false) {
                          // TODO handle cases for e.g. if the field is owned
                          (t, context)
                      case (None, None) =>
-                         logError(e, VariableUndefinedError(x, context.thisType.toString))
-                         (BottomType(), context)
+                         val tableLookup = context.contractTable.lookupContract(x)
+                         if (!tableLookup.isEmpty) {
+                           val ctTableOfConstructed = tableLookup.get
+                           if(ctTableOfConstructed.contract.isInterface)
+                                (InterfaceContractType(ctTableOfConstructed), context)
+                           else
+                               (BottomType(), context)
```

Comment:
What does this mean? Why bottom type here?

Reply:
Removed interface validation, as it is not relevant here.
- Added static validation in Checker instead